From 16a7b20524bfe923cc95f92a52d8e3e2fd4e3d8f Mon Sep 17 00:00:00 2001
From: Simon Ortet <simon.ortet@guidefitter.com>
Date: Tue, 11 Nov 2025 12:34:58 +0100
Subject: [PATCH] Remove uniqid from delayed messages for deduplication support

---
 Transport/Connection.php | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Transport/Connection.php b/Transport/Connection.php
index 3d2ad14..604fcbd 100644
--- a/Transport/Connection.php
+++ b/Transport/Connection.php
@@ -557,7 +557,7 @@ class Connection
                     'body' => $body,
                     'headers' => $headers,
                     // Entry need to be unique in the sorted set else it would only be added once to the delayed messages queue
-                    'uniqid' => $id,
+                    //'uniqid' => $id, This line needs to be commented out, we don't want uniqid because deduplication in the delay queue can't happen with it.
                 ]);
 
                 if (false === $message) {
@@ -601,12 +601,14 @@ class Connection
             throw new TransportException($error ?? $e->getMessage(), 0, $e);
         }
 
+        /*
+        // Removed because it would be triggered each time we add a message we already have in queue (deduplication) cause $added is false then
         if (!$added) {
             if ($error = $redis->getLastError() ?: null) {
                 $redis->clearLastError();
             }
             throw new TransportException($error ?? 'Could not add a message to the redis stream.');
-        }
+        }*/
 
         return $id;
     }
-- 
2.30.2

